### PR TITLE
🌟 Cb.get can return optional adminCodes field

### PR DESCRIPTION
**Pull request by astroash**
**changes:**
- Cb.get can return optional adminCodes field
- add test
- update cb type with optional field

### CODE REVIEW

File: src/api/v1/community_businesses/get.ts [hunk 1]
```diff
@@ -24,8 +25,18 @@ export default [
...
-    handler: async (request: Hapi.Request, h: Hapi.ResponseToolkit) => {
-      return Boom.notFound('Not implemented');
+    handler: async (request: GetCommunityBusinessesRequest, h: Hapi.ResponseToolkit) => {
+      const {
+        auth: { credentials: { role } } ,
+        server: { app: { knex } },
+        query,
+      } = request;
+
+      if (role !== RoleEnum.TWINE_ADMIN) {
```
Comment (eliascodes) [inline src/api/v1/community_businesses/get.ts]: Is this necessary? No other role (except `FUNDING_BODY_ADMIN`, which doesn't exist yet) has the `org_details-child` scope.

Comment (astroash) [reply src/api/v1/community_businesses/get.ts]: I think so. I want to ensure that we do not forget to add functionality for `FUNDING_BODY_ADMIN`.

Comment (eliascodes) [reply src/api/v1/community_businesses/get.ts]: If you just use the `isChildOrg` pre-req this is done for you.

File: src/models/community_business.ts [hunk 2]
```diff
@@ -144,7 +160,11 @@ export const CommunityBusinesses: CommunityBusinessCollection = {
         .innerJoin(
           'organisation',
           'organisation.organisation_id',
-          'community_business.organisation_id'),
+          'community_business.organisation_id')
+        .fullOuterJoin(
```
[the remark this answers is not in the document]
Comment (astroash) [reply src/models/community_business.ts]: Am I correct in thinking `fullOuterJoin` joins two tables and returns the values even if there isn't a match?

If so, then this is what I was after. If you use `innerJoin`, cbs without `volunteer_admin_code` table entries will not return with `CommunityBusinesses.get`. This is bad. I do not think that cbs _must_ have an admin code (even if we add them by default).

[the remark this answers is not in the document]
Comment (eliascodes) [reply src/models/community_business.ts]: I think what you want is `LEFT OUTER JOIN`. I always go back to [this](https://blog.codinghorror.com/a-visual-explanation-of-sql-joins/)

[the remark this answers is not in the document]
Comment (astroash) [reply src/models/community_business.ts]: I remember Saral showing me this article 😊. My description above is correct, but you're right, `LEFT OUTER JOIN` is more specific.

**Pull request merged**